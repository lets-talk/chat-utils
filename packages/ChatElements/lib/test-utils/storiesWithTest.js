import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
