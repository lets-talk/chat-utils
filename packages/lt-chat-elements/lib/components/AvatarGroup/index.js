import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Flex } from 'grid-styled';

import Avatar from '../Avatar';
import { PersonType } from '../../utils/types';
import { avatarGroupItem } from '../../utils/style';


const StyledTeamContainer = styled.div`
  width: 35px;
  height: 35px;
  position: relative;
  float: left;
`;

const AvatarGroup = (props) => {
  const {
    avatars, size, type, withStatus, groupType, xDistance, yDistance,
  } = props;

  const StyledAvatarGroupContainer = styled(Flex)`
    position: relative;
    left: 0;
    margin-right: ${() => groupType === 'line' ? avatars.length * 10 : 0}px;
  `;

  return (
    <StyledAvatarGroupContainer className={props.className}>
      <StyledTeamContainer>
        {avatars.map((person, index) => {
          const StyledAvatarContainer = styled.div`
              position: absolute;
              ${() => avatarGroupItem(index, avatars.length, groupType, xDistance, yDistance)}
              width: 100%;
              height: 100%;
          `;
          return (
            <StyledAvatarContainer key={person.id}>
              <Avatar
                size={size}
                src={person.avatar}
                status={person.status}
                withStatus={withStatus}
                type={type}
              />
            </StyledAvatarContainer>
          );
        })
        }
      </StyledTeamContainer>
    </StyledAvatarGroupContainer>
  );
};

AvatarGroup.defaultProps = {
  avatars: [],
  size: 'xsmall',
  xDistance: 12,
  yDistance: 12,
  type: 'circle',
  groupType: 'circle',
  withStatus: false,
  className: 'LT-AvatarGroup-Container',
};

AvatarGroup.propTypes = {
  /**
   * avatars: List of PersonType to show as avatars
   */
  avatars: PropTypes.arrayOf(PersonType),
  /**
   * How each avatar is showed. Valid types: default, circle, rounded(border radius 5px), flexible
   */
  type: PropTypes.oneOf(['default', 'circle', 'rounded', 'flexible']),
  /**
   * How avatars images are showed. Valid types: line, circle
   */
  groupType: PropTypes.oneOf(['line', 'circle']),
  /**
   * image size. default (25px), xsmall(30px), small(35px), medium(40px), large(45px), xlarge (55px)
   */
  size: PropTypes.string,
  /**
   * image xDistance. Distance on the x-axis between avatars
   */
  xDistance: PropTypes.number,
  /**
   * image yDistance. Distance on the y-axis between avatars
   */
  yDistance: PropTypes.number,
  /**
   * withStatus choose wheter or not show an status indicator.
   */
  withStatus: PropTypes.bool,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
};

export default AvatarGroup;
