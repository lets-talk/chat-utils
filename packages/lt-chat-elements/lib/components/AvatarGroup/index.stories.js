import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select, text, object, number } from '@storybook/addon-knobs';

// Component to show on storybook
import AvatarGroup from '.';
import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('AvatarGroup', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('AvatarGroup'));

const avatars = [
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
  {
    avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
    email: '',
    type: 'Agent',
    status: 'offline',
  },
];

stories.addWithInfo(
  'AvatarGroup',
  'This is the basic usage with the AvatarGroup with providing a label to show the text.',
  () => (
    <div
      style={{
        position: 'absolute', top: '50%', left: '50%', width: '100px', height: '100px', marginTop: '50px', marginLeft: '50px',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <AvatarGroup
          avatars={object('Avatars', avatars)}
          className={text('Classname')}
          type={select('Type', { default: 'default', rounded: 'rounded', circle: 'circle' }, 'circle')}
          groupType={select('Group Type', { circle: 'circle', line: 'line' }, 'circle')}
          size={select('Size', {
            xsmall: 'xsmall', small: 'small', medium: 'medium', large: 'large', xlarge: 'xlarge',
          }, 'xsmall')}
          xDistance={number('xDistance', 12, {
            range: true, min: 1, max: 100, step: 1,
          })}
          yDistance={number('yDistance', 12, {
            range: true, min: 1, max: 100, step: 1,
          })}
          withStatus={boolean('Show Status', false)}
        />
      </WrapWithTheme>
    </div>
  )
);
