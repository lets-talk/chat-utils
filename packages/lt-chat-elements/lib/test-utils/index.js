// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
