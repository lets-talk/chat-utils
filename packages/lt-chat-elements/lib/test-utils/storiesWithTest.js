import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
