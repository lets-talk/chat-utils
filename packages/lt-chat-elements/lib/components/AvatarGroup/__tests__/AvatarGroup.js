import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import AvatarGroup from '../index';
import { renderWithTheme } from '../../../test-utils';

describe('AvatarGroup component', () => {
  const props = {
    size: 'small',
    withStatus: true,
    avatars: [
      {
        id: 1,
        type: 'Agent',
        status: 'online',
        avatar: 'http://i46.tinypic.com/sexbb8.png',
        name: 'Sandino',
        alt: 'Agent image',
      },
      {
        id: 2,
        type: 'Client',
        status: 'online',
        avatar: 'http://i46.tinypic.com/sexbb8.png',
        name: 'Sandino',
        alt: 'Client Image',
      },
      {
        id: 3,
        type: 'Bot',
        status: 'online',
        avatar: 'http://i46.tinypic.com/sexbb8.png',
        name: 'BotName',
        alt: 'Bot Image',
      },
    ],
  };

  it('should render without data', () => {
    const component = shallow(<AvatarGroup />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a group in a circle (groupType: circle)', () => {
    const component = shallow(<AvatarGroup {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a group in a line (groupType: line)', () => {
    const newProps = { ...props, groupType: 'line' };
    const component = shallow(<AvatarGroup {...newProps} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Styled: should have the correct styles (Those that depends on the theme)', () => {
    const newProps = { ...props, groupType: 'line' };
    const tree = renderWithTheme(<AvatarGroup {...newProps} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('margin-right', '30px');
  });
});
