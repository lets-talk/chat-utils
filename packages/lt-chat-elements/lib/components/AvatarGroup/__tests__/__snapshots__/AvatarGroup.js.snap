// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup component Styled: should have the correct styles (Those that depends on the theme) 1`] = `
.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0, margin:0 auto;
  -ms-flex-negative: 0, margin:0 auto;
  flex-shrink: 0, margin:0 auto;
  overflow: hidden;
}

.c5 {
  width: undefined;
  height: undefined;
  position: relative;
}

.c6 {
  width: 100%;
  height: 100%;
  background-color: rgb(188,188,188);
  width: calc(calc( 5 * 8px) - 5px);
  height: calc(calc( 5 * 8px) - 5px);
  border-radius: 50%;
  vertical-align: middle;
}

.c7 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c8 {
  position: absolute;
  right: 0px;
  border-radius: 100%;
  padding: 1px;
  font-weight: 600;
  width: calc(calc( 5 * 8px) / 9);
  height: calc(calc( 5 * 8px) / 9);
  bottom: 0px;
  right: 0px;
  border: 1px solid #FFFFFF;
  background-color: rgb(67,160,71);
}

.c2 {
  width: 35px;
  height: 35px;
  position: relative;
  float: left;
}

.c0 {
  position: relative;
  left: 0;
  margin-right: 30px;
}

.c3 {
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
}

.c9 {
  position: absolute;
  top: 0px;
  left: 12px;
  width: 100%;
  height: 100%;
}

.c10 {
  position: absolute;
  top: 0px;
  left: 24px;
  width: 100%;
  height: 100%;
}

<div
  className="LT-AvatarGroup-Container c0 c1"
>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div
        className="LT-Avatar-Container c4"
        size="small"
      >
        <div
          className="c5"
        >
          <img
            alt=""
            className="c6"
            size="small"
            sizes={null}
            src="http://i46.tinypic.com/sexbb8.png"
            srcSet={null}
            type="circle"
          />
          <div
            className="c7"
          />
          <div
            className="c8"
            size="small"
          >
             
          </div>
        </div>
      </div>
    </div>
    <div
      className="c9"
    >
      <div
        className="LT-Avatar-Container c4"
        size="small"
      >
        <div
          className="c5"
        >
          <img
            alt=""
            className="c6"
            size="small"
            sizes={null}
            src="http://i46.tinypic.com/sexbb8.png"
            srcSet={null}
            type="circle"
          />
          <div
            className="c7"
          />
          <div
            className="c8"
            size="small"
          >
             
          </div>
        </div>
      </div>
    </div>
    <div
      className="c10"
    >
      <div
        className="LT-Avatar-Container c4"
        size="small"
      >
        <div
          className="c5"
        >
          <img
            alt=""
            className="c6"
            size="small"
            sizes={null}
            src="http://i46.tinypic.com/sexbb8.png"
            srcSet={null}
            type="circle"
          />
          <div
            className="c7"
          />
          <div
            className="c8"
            size="small"
          >
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup component should render a group in a circle (groupType: circle) 1`] = `
<StyledAvatarGroupContainer
  className="LT-AvatarGroup-Container"
>
  <StyledTeamContainer>
    <styled.div
      key="1"
    >
      <Avatar
        alt=""
        className="LT-Avatar-Container"
        color={null}
        size="small"
        sizes={null}
        src="http://i46.tinypic.com/sexbb8.png"
        srcSet={null}
        status="online"
        type="circle"
        withStatus={true}
      />
    </styled.div>
    <styled.div
      key="2"
    >
      <Avatar
        alt=""
        className="LT-Avatar-Container"
        color={null}
        size="small"
        sizes={null}
        src="http://i46.tinypic.com/sexbb8.png"
        srcSet={null}
        status="online"
        type="circle"
        withStatus={true}
      />
    </styled.div>
    <styled.div
      key="3"
    >
      <Avatar
        alt=""
        className="LT-Avatar-Container"
        color={null}
        size="small"
        sizes={null}
        src="http://i46.tinypic.com/sexbb8.png"
        srcSet={null}
        status="online"
        type="circle"
        withStatus={true}
      />
    </styled.div>
  </StyledTeamContainer>
</StyledAvatarGroupContainer>
`;

exports[`AvatarGroup component should render a group in a line (groupType: line) 1`] = `
<StyledAvatarGroupContainer
  className="LT-AvatarGroup-Container"
>
  <StyledTeamContainer>
    <styled.div
      key="1"
    >
      <Avatar
        alt=""
        className="LT-Avatar-Container"
        color={null}
        size="small"
        sizes={null}
        src="http://i46.tinypic.com/sexbb8.png"
        srcSet={null}
        status="online"
        type="circle"
        withStatus={true}
      />
    </styled.div>
    <styled.div
      key="2"
    >
      <Avatar
        alt=""
        className="LT-Avatar-Container"
        color={null}
        size="small"
        sizes={null}
        src="http://i46.tinypic.com/sexbb8.png"
        srcSet={null}
        status="online"
        type="circle"
        withStatus={true}
      />
    </styled.div>
    <styled.div
      key="3"
    >
      <Avatar
        alt=""
        className="LT-Avatar-Container"
        color={null}
        size="small"
        sizes={null}
        src="http://i46.tinypic.com/sexbb8.png"
        srcSet={null}
        status="online"
        type="circle"
        withStatus={true}
      />
    </styled.div>
  </StyledTeamContainer>
</StyledAvatarGroupContainer>
`;

exports[`AvatarGroup component should render without data 1`] = `
<StyledAvatarGroupContainer
  className="LT-AvatarGroup-Container"
>
  <StyledTeamContainer />
</StyledAvatarGroupContainer>
`;
